src/main.ts

```diff
@@ -29,19 +29,25 @@ export function activate(context: vscode.ExtensionContext): any {
 
     let logger = new Logger(text => _channel.append(text));
 
-    ensureRuntimeDependencies(extension, logger, reporter)
-        .then((success : boolean) => {
-            // activate language services
-            OmniSharp.activate(context, reporter, _channel);
-
-            // register JSON completion & hover providers for project.json
-            context.subscriptions.push(addJSONProviders());
-            
-            if (success) {
-                // activate coreclr-debug
-                coreclrdebug.activate(extension, context, reporter, logger, _channel);
-            }
-        });
+    let runtimeDependenciesExist = await ensureRuntimeDependencies(extension, logger, reporter);
+
+    if (!runtimeDependenciesExist) {
+        //do something
```

Comment:
I don't think you want to do anything else in this function if `runtimeDependenciesExist` is false.

Reply:
throw new Error("dependencies don't exist");
?

Since we're just starting to use async/await in this repo, I wanted to hear what folks think...

Reply:
If you `throw`, what will VS Code do?

Reply:
I'm pretty sure VS Code handles it and displays a message. https://github.com/Microsoft/vscode/blob/27c07934d7d538f6799dc78dc19a174f8e31aaa7/src/vs/workbench/api/node/extHostExtensionActivator.ts#L314-L320

Reply:
In this case I think you do NOT want to throw as we already tell the user what failed. Maybe expose a bool back to the test that the test can check?

Reply:
This is not present for test purposes. Once I de-thenefied this block of code it became apparent that we have forked behaviors. Prior to this change:
 - IF ensureRuntimeDependencies == false
  - Omnisharp tries to activate anyways. Is that legal?
  - JsonProviders are registered. Again, is that legal?
  - coreclrdebug does not activate

This seemed funny to me. Is it the intent that Omnisharp starts happily but the debugger is not activated? Shouldn't we behave like a single extension?

Reply:
Registering the Json providers is fine.
Starting OmniSharp could be fine if the issue was that the debugger failed to download. I think that is why that code always runs today.
We definitely do NOT want to activate coreclrdebug, as that code is there to finalize the debugger install, but it doesn't make sense to do so if the download failed.

Reply:
So are we saying that `ensureRuntimeDependencies` only pulls down debugger dependencies and that it has no effect on omnisharp and json provider?

While writing these integration tests I've already discovered an undefined state in the control [The control claims to be active even though it's not ready to receive commands]. This feels like another place where we end up in an undefined state. If we're confident that `ensureRuntimeDependencies` only concerns the debugger then we can design/test for that scenario. Otherwise, we are missing an opportunity to direct the user to fix their installation and are instead providing them with an undefined experience...

Reply:
`ensureRuntimeDependencies` will download OmniSharp and the Debugger.

What was the case you ran into?

I know that `completeDebuggerInstall` is really async so if we want to wait for the debugger install to be complete, there should be an `await` on src/coreclr/activate.ts line 51.

Reply:
@gregg-miskelly the issue I ran into is that I issued a command registered by our extension before omnisharp initialized and that command invocation resulted in an error.

I opened https://github.com/OmniSharp/omnisharp-vscode/issues/1840 to track startup behavior. Let's decide what we want to do in that issue. I'll remove the current `do something` block for this PR.